smif/convert/interval.py

```diff
@@ -360,6 +359,8 @@ class IntervalSet(ResolutionSet):
 
     def __init__(self, name, data, base_year=2010):
         self._name = name
+        self._filename = ''
```

Comment:
You're right in the pull request comment - filename is definitely data-layer-implementation-specific.

If we have some unique identifier for each object (name) then it might be reasonable for the data layer to maintain that mapping from file to object.

Otherwise if it's simpler to hang on to the metadata, perhaps `filename` could be renamed to something more generic (data_source? store_location?). What do you reckon?